File: examples/key/new_local_certificate/new_local_certificate.ts
import {
  KeyClient,
  KeyType,
  LocalCertificateParams,
  SubjectCertificateParams,
} from '@bloock/sdk';

const keyClient = new KeyClient();

// Load a managed key
const _certificate = await keyClient.newLocalCertificate(
  new LocalCertificateParams(
    KeyType.Rsa2048,
    new SubjectCertificateParams('Bloock'),
    'password',
    2
  )
);

File: examples/key/new_local_certificate/NewLocalCertificate.java
import com.bloock.sdk.client.KeyClient;
import com.bloock.sdk.entity.key.*;

public class NewLocalCertificate {
    public static void main(String[] args) throws Exception {
        KeyClient keyClient = new KeyClient();

        // Create a local certificate
        LocalCertificate certificate = keyClient.newLocalCertificate(new LocalCertificateParams(
                KeyType.Rsa2048,
                new SubjectCertificateParams(
                        "BLOOCK",
                        "Organization",
                        "Organization Unit",
                        "Location",
                        "State",
                        "Country"),
                "password",
                12));
    }
}